rdr_service/dao/genomics_dao.py

```diff
@@ -4250,3 +4251,39 @@ def from_client_json(self):
     def get_id(self, obj):
         pass
 
+
+class GenomicAppointmentEventMetricsDao(UpdatableDao, GenomicDaoMixin):
+    def __init__(self):
+        super(GenomicAppointmentEventMetricsDao, self).__init__(
+            GenomicAppointmentEventMetrics, order_by_ending=['id']
+        )
+
+    def from_client_json(self):
+        pass
+
+    def get_id(self, obj):
+        return obj.id
+
+    def get_missing_appointments(self):
+        with self.session() as session:
+            return session.query(
+                GenomicAppointmentEventMetrics.id,
+                GenomicAppointmentEventMetrics.event_type,
+                GenomicAppointmentEventMetrics.event_authored_time,
+                GenomicAppointmentEventMetrics.participant_id,
+                GenomicAppointmentEventMetrics.appointment_event,
+                GenomicAppointmentEventMetrics.reconcile_job_run_id,
+                GenomicAppointmentEvent.id.label('appointment_event_id')
+            ).outerjoin(
+                GenomicAppointmentEvent,
+                and_(
+                    GenomicAppointmentEvent.participant_id == GenomicAppointmentEventMetrics.participant_id,
+                    GenomicAppointmentEvent.module_type == GenomicAppointmentEventMetrics.module_type,
+                    GenomicAppointmentEvent.event_type == GenomicAppointmentEventMetrics.event_type,
+                    GenomicAppointmentEvent.event_authored_time == GenomicAppointmentEventMetrics.event_authored_time
```

Comment:
Just curious, do you want to join on appointment ID here too, or is event_authored_time sufficient? Also, you may want to index it.

Reply:
I think `event_authored_time` is sufficient tbh in this case. I am using it because its actually the most granular attribute that I can use for reconciliation since each one of the json objects lacks a unique id on them.

Reply:
Sounds good.